.github/workflows/ci.yml

```diff
@@ -22,8 +26,39 @@ jobs:
     - name: Install Newman dependency
       run: npm install --prefix tests/postman/
 
-    - name: Run the API and Postman's tests
-      run: |
-        cd tests/postman/ && newman run *
+    - name: Start Netlify Dev for local function testing
+      run:  npm run netlify-dev & sleep 10 && cd tests && node newman.js 
+      env:
+        STRIPE_TEST_KEY: ${{ secrets.STRIPE_TEST_KEY }}
+        ENDPOINT: http://localhost:5050
+
+    - name: Run tests against deployed preview
+      run:   ./scripts.sh wait-for-url https://deploy-preview-2--ambianic-serverless.netlify.app && cd ./tests && node newman.js 
       env:
-        CI: true
+        STRIPE_TEST_KEY: ${{ secrets.STRIPE_TEST_KEY }}
+        ENDPOINT: https://deploy-preview-${{github.event.number}}--ambianic-serverless.netlify.app/.netlify/functions
+    
+    - name: Generate OpenAPI specs
+      run: |
+        npm run generate-swagger && ./scripts.sh cleanup
+    
+    - name: Create a static API documentation page from openapi specs
+      run: npm run generate-docs
+
+      # run this step only if its a push to the main branch
+    - name: Publish static API documentation to Github Pages
+      if: ${{ github.event_name == 'push' }}
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        personal_token: ${{ secrets.REPOSITORY_ACCESS_TOKEN }}
+        publish_dir: ./github-pages
+
+    - name: Run the semantic-release
```

Comment:
@vickywane where is the semantic release script. This just runs the command but what does it do? Where is the versioning, changelog, packaging and publishing code that semantic release expects?

Reply:
Semantic-release is currently being made to be **zero-config** as possible for quick adoption.

Please note the following:

- Semantic-release is configured in the `release` section of the `package.json` file. ( See docs reference [here](https://semantic-release.gitbook.io/semantic-release/usage/configuration#configuration-file) )

- Versioning, changelog, and packaging are done through the use of plugins within the `release` object. ( See docs reference [here](https://semantic-release.gitbook.io/semantic-release/usage/configuration#plugins) )

- Semantic-release by default would push to the `repository.url` field in the `package.json` file. ( See docs reference [here](https://semantic-release.gitbook.io/semantic-release/usage/configuration#repositoryurl) )

- To publish, semantic-release would make use of the GITHUB_TOKEN and NPM_TOKEN to authenticate and publish releases to npm except if the `npmPublish` field is set to false.